### DISCUSSION TITLE
First release candidate of Jamulus 3.8.1

### DISCUSSION THREAD

**Discussion by gilgongo** [jamulussoftware/jamulus#2062] [Announcements]
We have just released version 3.8.1rc1 of Jamulus and would be happy for you to download and test out this version. If all is well, this will become the 3.8.1 release.

You can view the Release Notes and download versions for Windows, Mac, Linux (Debian/Ubuntu), Android (still experimental) and source code here:

https://github.com/jamulussoftware/jamulus/releases/tag/r3_8_1rc1.

If you find any issues with it, please [raise an Issue](https://github.com/jamulussoftware/jamulus/issues) with as many details as possible.

### UPDATE

A German translation has now been added to a second release candidate r3_8_1rc2.

**Comment by jujudusud** [comment]
I build this RC1 on ArchLinux, everything is ok from my point of view.

**Comment by rdica** [comment]
RC1 builds/runs fine on Fedora 33, both as server and client.

**Comment by softins** [comment]
One significant feature in this release is support for IPv6 in direct connections to a compatible server. See [here](https://github.com/jamulussoftware/jamulus/pull/1938#issue-958235326) for details.

**Reply by rdica** [reply]
Yes, thats one of the prime reasons I'm testing both as client and server :)

**Reply by rdica** [reply]
I played about 5 hours last night on my server with a direct IPv6 connection, seemed solid, same latency as I get from IPv4, perhaps a few ms lower. 12 hops to my server over v6, 14 over v4. I'm pleased overall.

**Reply by softins** [reply]
Great, thanks for the feedback!

**Comment by 38github** [comment]
I tested it on Debian 11 amd64 connected to a server running it also and during my limited testing I found no issues.

**Comment by ghost** [comment]
**Found Warning while compiling:**
src/socket.cpp: In member function ‘void CSocket::SendPacket(const CVector<unsigned char>&, const CHostAddress&)’:
src/socket.cpp:328:13: warning: ‘status’ may be used uninitialized in this function [-Wmaybe-uninitialized]
             if ( status >= 0 )
             ^
**Fix warning**
with (line 256):
int status = 0;
instead of:
int status;

**Reply by softins** [reply]
I found this once when compiling on CentOS 7 with GCC 4.8.5. It doesn't happen on my Pi with GCC 8.3.0, or Ubuntu 18 with GCC 7.5.0. My guess is the newer GCCs can understand the code paths a bit better. But I'll add the initialisation for tidiness.

**Reply by ghost** [reply]
I just read the man page for sendto() and found that it would be better to initialise:
int status = -1; // default to failure if sendto() is not called.
(my bad for assuming 0 was ok)

man page for sendto():
On  success,  these  calls  return  the number of characters sent. 
On error, -1 is returned, and errno is set appropriately.

**Reply by ghost** [reply]
A further correction to avoid ambiguity of the -1 error status of sendto():
int status = -2; // create a local -2 failure if sendto() is not called.

**Reply by softins** [reply]
In this case, I think it’s best as it is. In the almost-impossible case of the code taking the path that doesn’t call `status = sendto()`, there is no point in retrying, as the same would happen again, so we may as well break out of the loop anyway.
